libs/server-sdk/src/data_systems/lazy_load/lazy_load_system.hpp

```diff
@@ -72,16 +72,25 @@ class LazyLoad final : public data_interfaces::IDataSystem {
     void RefreshSegment(std::string const& key) const;
 
     template <typename TResult>
-    static TResult Get(data_components::ExpirationTracker::TrackState state,
-                       std::function<void(void)> const& refresh,
-                       std::function<TResult(void)> const& get) {
+    TResult Get(std::string const& key,
```

Comment:
The algorithm isn't as elegant now as it was with the fallthroughs, but I wanted to log the actual action taking place. Suppose I could use a string stream instead.

Reply:
A little sad, it was relatively pleasant looking for C++. :-P

Reply:
You're right, it was a jewel. Gonna try to fix it up with a string-stream.

Reply:
Elegance restored.